telemetry/specs/INT.mdk

```diff
@@ -494,19 +494,117 @@ corruption at preceding hops.
 
-1. *INT over TCP/UDP* - A shim header is inserted following TCP/UDP
+1. "INT over IPv6" - INT Headers are carried in the IPv6 packets as Hop-by-Hop option
+or Destination option.
+2. *INT over TCP/UDP* - A shim header is inserted following TCP/UDP
 header. INT Headers are carried between this shim header and TCP/UDP payload.
 This approach doesn’t rely on any tunneling/virtualization mechanism and is
 versatile to apply INT to both native and virtualized traffic.
-2. *INT over VXLAN* - VXLAN generic protocol extensions
+3. *INT over VXLAN* - VXLAN generic protocol extensions
 (draft-ietf-nvo3-vxlan-gpe) are used to carry INT Headers between
 the VXLAN header and the encapsulated VXLAN payload.
-3. *INT over Geneve* - Geneve is an extensible tunneling framework, allowing
+4. *INT over Geneve* - Geneve is an extensible tunneling framework, allowing
 Geneve options to be defined for INT Headers.
 
+### INT over IPv6
+
+In case the traffic being monitored is not encapsulated by any virtualization
+header, INT over VXLAN or INT over Geneve is not helpful. Instead, 
+the INT metadata can be carried as Hop-by-Hop option or Destination
+option data. 
+
+IPv6 Hop-by-Hop and Destination Option format for carrying INT Header and 
+Metadata:
+
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+|    Nxt HDR    |  HDR Ext Len  |  Option Type  |  Opt Data Len |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  |	  |	
+|         Reserved (MBZ)        |  Variable Opt Data (INT Data) |  I
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  N
+.                Variable Opt Data (INT Data)                   .  T
+.                                                               .  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  |
+|  Variable Opt Data (INT Data) |         Padding               |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+ 
+
+Nxt Hdr: 8-bit selector. Identifies the type of header immediately following the 
+Hop-by-Hop or Desitnation Options header. Uses the same values as the IPv4 Protocol
+Field [IANA-PN]
+
+HDR Ext Len: 8-bit unsigned integer. Length of the Hop-by-Hop or Desitnation
+Options header in 8-octet units, not including the first 8 octets.
+
+Option Type: 8-bit identifier of the type of option.
+
+  001xxxxxx 8-bit identifier of the type of option. xxxxxx=TBD_IANA_INT_HOP_BY_HOP_OPTION_IPV6.
+  001xxxxxx 8-bit identifier of the type of option. xxxxxx=TBD_IANA_INT_DESTINATION_OPTION_IPV6.
+
+Opt Data Len: 8-bit unsigned integer. Length of the Reserved and Option Data field of this 
+option, in octets.
+
+Reserved (MBZ): 16 bit field, must be filled with zeroes.
+Variable Opt Data: INT Header and Metadata, multiple of four octets in length.
+
+Padding: 16-bit pad. Needed to ensure that the variable length of the complete 
+Hop-by-Hop or Destination Options Header is an integer multiple of 8 octets long.
```

Comment:
For v2, it looks like we have 4 octets shim header and 12 octets fixed part of INT header before the variable metadata. There are two options with respect to padding:

1. Go with 4n alignment. When INT is the only option present in the hop-by-hop options, then the "Padding|(MBZ)" above would be inserted in front of this option, and if Hop ML is even then there would be an additional 4 octets of padding after this option. If Hop ML is odd, then the additional 4 octets of padding after this option would only be present at even hops, so each hop either strips the incoming 4 octets of padding after this option, or adds 4 octets of padding after this option. If there are other hop-by-hop options present, padding may differ both before and after this option.
2. Go with 8n alignment and require that Hop ML must be even. When INT is the only option present in the hop-by-hop options, then the "Padding|(MBZ)" in front of this option would be 6 octets rather than 2 octets. No padding would be added after this option. If there are other hop-by-hop options present, then the padding may differ both before and after this option, but it would not need to be adjusted at each hop.

IOAM has a similar issue, but the header size is different so the 4n alignment implications are different. When IOAM incremental trace is the only option present in the hop-by-hop options, and if hop ML is even then there would be no padding after this option. 8n alignment would not work since the total size of the fixed headers is not a multiple of 8 octets.